target/linux/ipq40xx/image/generic.mk

```diff
@@ -606,7 +606,7 @@ define Device/meraki_mr33
 	SOC := qcom-ipq4029
 	BLOCKSIZE := 128k
 	PAGESIZE := 2048
-	DEVICE_PACKAGES := -swconfig ath10k-firmware-qca9887-ct
+	DEVICE_PACKAGES := -swconfig ath10k-firmware-qca9887-ct kmod-leds-gpio
```

Comment:
kmod-leds-gpio should come from the main Makefile's `DEVICE_PACKAGES`:
https://github.com/openwrt/openwrt/blob/f5d071a508b3e3d218b8ed6f6f5a999384bb7f6d/target/linux/ipq40xx/Makefile#L19

It's strange that the MR33 would not have that?! (Since most other ipq40xx devices also use leds-gpio)

Reply:
That is strange, but you can actually deselect them in menuconfig.

Reply:
I noticed this, even if it is included in the device Makefile, it can still be unchecked

Reply:
When using "multiple devices" profile, then this module is actually selected at least with "M".

Reply:
So, question then: Does the reset button work or not? 

It's possible to just deselect kmod-leds-gpio if you only built the MR33. 
For the "mutliple devices" it should always be at least either module or built-in.

Reply:
Note that DEVICE_PACKAGES requires CONFIG_TARGET_PER_DEVICE_ROOTFS=y to be evaluated if you built with "multiple devices".

Reply:
Many targets have directly built-in CONFIG_LEDS_GPIO, maybe ipq40xx can also be adjusted to this

Reply:
@nasbdh9 Oh, let me explain my position: I smell that something is not right here. For all accounts, kmod-gpio-leds should be there in the default config... together with all the other package specified in the main `DEVICE_PACKAGES`. That's why I'm so curious about this. This PR looks to me like a bug-report in disguise. If there's a underlying problem with the infrastructure build code/rules, I would like to know this. Because if this is widespread other devices be affected too, right?

On the other hand, it's possible to just deselect CONFIG_PACKAGE_kmod-leds-gpio for custom builds. I think from an user's perspective this could be either fully intended or an accident.  Should we take away their choice by forcing CONFIG_LEDS_GPIO in config-default?

Reply:
> @nasbdh9 Oh, let me explain my position: I smell that something is not right here. For all accounts, kmod-gpio-leds should be there in the default config... together with all the other package specified in the main `DEVICE_PACKAGES`. That's why I'm so curious about this. This PR looks to me like a bug-report in disguise. If there's a underlying problem with the infrastructure build code/rules, I would like to know this. Because if this is widespread other devices be affected too, right?

Almost like this

> 
> On the other hand, it's possible to just deselect CONFIG_PACKAGE_kmod-leds-gpio for custom builds. I think from an user's perspective this could be either fully intended or an accident. Should we take away their choice by forcing CONFIG_LEDS_GPIO in config-default?

Yes, I think the default configuration should include CONFIG_LEDS_GPIO instead of being selected as a module by default
This is also the case for several other targets. Can anyone explain why it defaults to a module instead of the default configuration?

Reply:
? So there's no issue here with kmod-leds-gpio?

Well fine.